verification.py

```diff
@@ -5919,6 +5919,63 @@ def test_bottleneck_coalescence_time_T2(self):
         self._run(2)
 
 
+class MicrosatAnalytical(Test):
+    """
+    Analytical comparisons wrt to microsatellites
+    """
+
+    def test_SMM_variance(self):
+        """
+        Under the strict SMM model we expect the
+        Variance in repeat number of a sample
+        to be equal to 4Nu. This result has
+        been derived a number of times but a good
+        citation is Goldstein et al (1995)
+        https://doi.org/10.1093%2Fgenetics%2F139.1.463
+        """
+
+        def copy_number_matrix(ts):
+            """
+            Returns the copy number matrix from the specified tree sequence
+            simulated under a MicrosatMutationModel.
+            """
+            C = np.zeros((ts.num_sites, ts.num_samples), dtype=int)
+            for var in ts.variants():
+                alleles = np.array([int(allele) for allele in var.alleles])
+                C[var.site.id] = alleles[var.genotypes]
+            return C
+
+        rates = np.linspace(1e-5, 1e-2, num=10)
+        avg_variances = []
+        expected = []
+        N = 1000
+        nsamp = 100
+        nreps = 100
+        mod = msprime.SMM(lo=3, hi=500)
+        for r in rates:
+            reps = msprime.sim_ancestry(
+                ploidy=2,
+                samples=nsamp,
+                num_replicates=nreps,
+                population_size=N,
+                sequence_length=1,
+            )
+            variances = []
+            for ts in reps:
+                mts = msprime.sim_mutations(ts, rate=r, model=mod)
+                C = copy_number_matrix(mts)
+                variances.append(np.var(C.flatten(), ddof=1))
+            avg_variances.append(np.mean(variances))
+            expected.append(2 * N * r)
```

Comment:
factor of 2 off here?

Reply:
Maybe easier to follow if we set ``ploidy=1`` in sim_ancestry and compared with ``4Nr``?

Reply:
see below comment